services/catalog/src/simcore_service_catalog/services/director.py

```diff
@@ -85,29 +66,44 @@ def _process(cls, resp: Response) -> Optional[Dict]:
             msg = error or resp.reason_phrase
             raise HTTPException(resp.status_code, detail=msg)
 
-        return data
+        return data or {}
+
+    @functools.wraps(request_func)
+    async def request_wrapper(self: "AuthSession", path: str, *args, **kwargs):
```

Comment:
what does it mean with an annotation in form of a string?

Reply:
AuthSession is still not defined (see in the file is defined above).

In 3.6 this is the nly way to annotate a class that is still undefined

See e.g. this discussion
https://stackoverflow.com/questions/15853469/putting-current-class-as-return-type-annotation

Reply:
I see, thanks. let's also update python.